src/Model/Item.php

```diff
@@ -1740,6 +1676,68 @@ public static function insert($item, $notify = false, $dontcache = false)
 			return 0;
 		}
 
+		if ($item['gravity'] !== GRAVITY_PARENT) {
+			$toplevel_parent = self::getTopLevelParent($item);
+			if (empty($toplevel_parent)) {
+				return 0;
+			}
+
+			// If the thread originated from this node, we check the permission against the thread starter
+			$condition = ['uri' => $toplevel_parent['uri'], 'wall' => true];
+			$localTopLevelParent = self::selectFirst(['uid'], $condition);
+			if (!empty($localTopLevelParent['uid']) && !self::isAllowedByUser($item, $localTopLevelParent['uid'])) {
+				return 0;
```

Comment:
We should add some log message here, I think.

Reply:
Log messages are sent by `isAllowedByUser` in all the `return false` cases.

I know it feels weird, I've wondered this multiple times myself but it's already handled.

Reply:
But in this log messages we can't see if a post is rejected because the current user or the thread owner had blocked the author. I would prefer such a log messages especially to be able to check if this works at all. It's good for support requests, I think.

Reply:
The particular user id is included in the logger context.